### DISCUSSION TITLE
Why they are not the same by an same PromQL on different time.

### DISCUSSION THREAD

**Discussion by Sunyelw** [prometheus/prometheus#10350] [Q&A]
Why they are not the same and so different?

<img width="657" alt="image" src="https://user-images.githubusercontent.com/31758533/154656059-ab52c95d-8616-44a2-b0c7-90e500ceee13.png">

**Comment by Sunyelw** [comment]
and 1d is success
![image](https://user-images.githubusercontent.com/31758533/157580731-cd0d73d3-d471-488a-9091-4682aa63fe5c.png)


more than 1d no data
![image](https://user-images.githubusercontent.com/31758533/157580746-02e261d9-b556-406a-b3ff-0d185ec96d0d.png)

**Comment by Sunyelw** [comment]
@codesome Could you give me some help? Thx!

**Comment by codesome** [comment]
When the time of query is changed, the data that the query is running on also changes. Hence the result can be different.

Regarding the no data part, do you have data that is 2d old?

**Reply by Sunyelw** [reply]
No, the data 2d old had deleted by me for test.

**Reply by codesome** [reply]
You answered your own question. You deleted the data, hence you get no result.

**Reply by Sunyelw** [reply]
But why result of sum(increase[1w]) is so different? That's what really confuses me.
